### PR TITLE
[SPARK-42426][CONNECT] Fix DataFrameWriter.insertInto to call the corresponding method instead of saveAsTable

### PR DESCRIPTION
**Pull request by ueshin**
### What changes were proposed in this pull request?

Fixes `DataFrameWriter.insertInto` to call the corresponding method instead of `saveAsTable`.

### Why are the changes needed?

Currently `SparkConnectPlanner` calls `saveAsTable` instead of `insertInto` even for `DataFrameWriter.insertInto` in Spark Connect, but they have different logic internally, so we should use the corresponding method.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Enabled related tests.

### CODE REVIEW

**Review by HyukjinKwon — APPROVED** [review]

File: connector/connect/common/src/main/protobuf/spark/connect/commands.proto [hunk 1]
```diff
@@ -91,6 +91,16 @@ message WriteOperation {
   // (Optional) A list of configuration options.
   map<string, string> options = 9;
 
+  message SaveTable {
+    string table_name = 1;
+    TableSaveMethod save_method = 2;
+
+    enum TableSaveMethod {
+      SAVE_AS_TABLE = 0;
+      INSERT_INTO = 1;
```
Comment (zhengruifeng) [inline connector/connect/common/src/main/protobuf/spark/connect/commands.proto:100]: what about adding `SAVE_MODE_ INSERT_INTO` into existing `enum SaveMode`?

Comment (ueshin) [reply connector/connect/common/src/main/protobuf/spark/connect/commands.proto:100]: I'm afraid I don't think it's a good idea. The method and mode are orthogonal concepts. Spark Connect should let Spark decide how it should behave with the options.

Comment (zhengruifeng) [reply connector/connect/common/src/main/protobuf/spark/connect/commands.proto:100]: Got it, Thanks for the clarification!

File: connector/connect/common/src/main/protobuf/spark/connect/commands.proto [hunk 2]
```diff
@@ -91,6 +91,16 @@ message WriteOperation {
   // (Optional) A list of configuration options.
   map<string, string> options = 9;
 
+  message SaveTable {
+    string table_name = 1;
+    TableSaveMethod save_method = 2;
+
+    enum TableSaveMethod {
```
Comment (zhengruifeng) [inline connector/connect/common/src/main/protobuf/spark/connect/commands.proto:98]: last thing: according to style guide (which is used in connect protos) https://protobuf.dev/programming-guides/style/#enums , this `enum` start with `_UNSPECIFIED`

```
enum TableSaveMethod {
    TABLE_SAVE_MODE_UNSPECIFIED = 0;

}
```

**Review by zhengruifeng — APPROVED** [review]

File: connector/connect/common/src/main/protobuf/spark/connect/commands.proto (hunk 2 above)
Comment (ueshin) [reply connector/connect/common/src/main/protobuf/spark/connect/commands.proto:98]: Sure, will address it soon. Thanks!

File: connector/connect/common/src/main/protobuf/spark/connect/commands.proto [hunk 3]
```diff
@@ -91,6 +91,17 @@ message WriteOperation {
   // (Optional) A list of configuration options.
   map<string, string> options = 9;
 
+  message SaveTable {
+    string table_name = 1;
```
Comment (amaliujia) [inline connector/connect/common/src/main/protobuf/spark/connect/commands.proto:95]: Can you follow https://github.com/apache/spark/blob/master/connector/connect/docs/adding-proto-messages.md to add `(Required)` or `(Optional)` with explanation on the fields?

**Review by amaliujia — COMMENTED** [review]
LGTM with a minor comment to improve proto code style.

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.4.

**Pull request closed without merging**